docs/README.md

```diff
@@ -31,39 +106,176 @@ Install Mirascope and start building with LLMs in minutes.
 
 ```python
-from mirascope import OpenAIChat, Prompt
+from mirascope import Prompt
 
-class BookRecommendationPrompt(Prompt):
+class GreetingsPrompt(Prompt):
     """
-    Can you recommend some books on {topic}?
+    Hello! It's nice to meet you. My name is {name}. How are you today?
     """
 
-    topic: str
+    name: str
 
-prompt = BookRecommendationPrompt(topic="coding")
-print(str(prompt))
+prompt = GreetingsPrompt(name="William Bakst")
+
+print(GreetingsPrompt.template())
+#> Hello! It's nice to meet you. My name is {name}. How are you today?
+
+print(prompt)
+#> Hello! It's nice to meet you. My name is William Bakst. How are you today?
+```
+
+<details>
+<summary>Example of autocomplete and inline errors:</summary>
+<ul>
+    <li>Autocomplete:</li>
+    <img width="822" alt="Screenshot 2024-01-30 at 7 10 11 PM" src="https://github.com/Mirascope/mirascope/assets/99370834/812a97c1-7f0d-472c-842d-305d7f5da085">
+    <li>Inline Errors:</li>
+    <img width="571" alt="Screenshot 2024-01-30 at 7 08 54 PM" src="https://github.com/Mirascope/mirascope/assets/99370834/e09466a0-0941-46ec-abdb-e55cf84f1f25">
+</ul>
+</details>
+
+You can access the docstring prompt template through the `GreetingsPrompt.template()` class method, which will automatically take care of removing any additional special characters such as newlines. This enables writing longer prompts that still adhere to the style of your codebase:
+
+```python
+class GreetingsPrompt(Prompt):
+    """
+    Salutations! It is a lovely day. Wouldn't you agree? I find that lovely days
+    such as these brighten up my mood quite a bit. I'm rambling...
 
-model = OpenAIChat(api_key="YOUR_OPENAI_API_KEY")
-res = model.create(prompt)
-print(str(res))
+    My name is {name}. It's great to meet you.
+    """
+
+prompt = GreetingsPrompt(name="William Bakst")
+print(prompt)
+#> Salutations, good being! It is a lovely day. Wouldn't you agree? I find that lovely days such as these brighten up my mood quite a bit. I'm rambling...
+#> My name is William Bakst. It's great to meet you.
 ```
 
+The `str` method is written such that it only formats properties that are templated. This enables writing more complex properties that rely on one or more provided properties:
+
+```python
+class GreetingsPrompt(Prompt):
+    """
+    Hi! My name is {formatted_name}. {name_specific_remark}
+
+    What's your name? Is your name also {name_specific_question}?
+    """
+
+    name: str
+
+    @property
+    def formatted_name(self) -> str:
+        """Returns `name` with pizzazz."""
+        return f"⭐{self.name}⭐"
+
+    @property
+    def name_specific_question(self) -> str:
+        """Returns a question based on `name`."""
+        if self.name == self.name[::-1]:
+            return "a palindrome"
+        else:
+            return "not a palindrome"
+
+    @property
+    def name_specific_greeting(self) -> str:
```

Comment:
typo, method says greeting but property is called remark

Reply:
Done.